src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -94,9 +104,72 @@ public void Initialize(IFederationHistory federationHistory, IIdleFederationMemb
             this.nodeStats.RegisterStats(this.AddComponentStats, StatsType.Component, this.GetType().Name, 1200);
 
             this.isInitialized = true;
+
             this.logger.LogDebug("VotingManager initialized.");
         }
 
+        /// <summary> Remove all polls that started on or after the given height.</summary>
+        /// <param name="height">The height to clean polls from.</param>
+        public void DeletePollsAfterHeight(int height)
+        {
+            this.logger.LogInformation($"Cleaning poll data from height {height}.");
+
+            var idsToRemove = new List<int>();
+
+            this.polls = this.pollsRepository.GetAllPolls();
+
+            foreach (Poll poll in this.polls.Where(p => p.PollStartBlockData.Height >= height))
+            {
+                idsToRemove.Add(poll.Id);
+            }
+
+            if (idsToRemove.Any())
+            {
+                this.pollsRepository.DeletePollsAndSetHighestPollId(idsToRemove.ToArray());
+                this.polls = this.pollsRepository.GetAllPolls();
+            }
+        }
+
+        /// <summary> Reconstructs voting and poll data from a given height.</summary>
+        /// <param name="height">The height to start reconstructing from.</param>
+        public void ReconstructVotingDataFromHeight(int height)
+        {
+            try
+            {
+                this.isBusyReconstructing = true;
```

Comment:
This can get set twice right?

Reply:
Good point, I will the change.

Reply:
I've changed this so that entry into the `Reconstruct` method is checked so that we dont kick off the processing again. Dont really need a lock then 👍